engine/collection/ingest/engine.go

```diff
@@ -224,6 +225,13 @@ func (e *Engine) onTransaction(originID flow.Identifier, tx *flow.TransactionBod
 	// or a different cluster)
 	if originID == e.me.NodeID() {
 
+		// if the target cluster has no collection node other than possibly this
+		// collection node then don't attempt to propagate the transaction further
+		recipientIDs := txCluster.Filter(filter.Not(filter.HasNodeID(e.me.NodeID())))
```

Comment:
Is this potentially expensive?

Reply:
this will be `O(n)` where n is the number of nodes in a cluster..so roughly arnd 33 entries (for ~100 collection node and 3 cluster)

Reply:
It's certainly not optimized but I don't think it will be significantly expensive. It's linear in the size of the cluster (~100) and it allocates one map.